airframe-http-recorder/src/main/scala/wvlet/airframe/http/recorder/HttpRecorder.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe.http.recorder
+
+import java.net.URI
+
+import com.twitter.finagle.{Http, Service}
+import com.twitter.finagle.builder.ClientBuilder
+import com.twitter.finagle.http.{Request, Response, Status}
+import com.twitter.finagle.service.RetryPolicy
+import com.twitter.util.Future
+import javax.net.ssl.SSLContext
+import wvlet.airframe.http.finagle.{FinagleServer, FinagleServerConfig}
+import wvlet.log.LogSupport
+import wvlet.log.io.IOUtil
+
+case class HttpRecorderConfig(destUri: String,
+                              sessionName: String = "default",
+                              folder: String = "fixtures",
+                              // Drop the session records for recording mode
+                              dropSessionIfExists: Boolean = true,
+                              recordTableName: String = "record",
+                              // Specify the port to use. The default is finding an available port
+                              private val port: Int = -1,
+                              // A filter for customizing HTTP request headers to use for generating database keys.
+                              // For example, we should remove headers that depends on the current time, etc.
+                              headerExcludes: String => Boolean = HttpRecorder.defaultHeaderExclude,
+                              fallBackHandler: Service[Request, Response] = HttpRecorder.defaultFallBackHandler) {
+
+  def sqliteFilePath  = s"${folder}/${sessionName}.sqlite"
+  lazy val serverPort = if (port == -1) IOUtil.unusedPort else port
```

Comment:
Just curious, why `sqliteFilePath` is a method despite `serverPort` and `destHostAndPort` are `lazy val`?

Reply:
Both can be val. IOUtil.unusedPort will return an available port at the moment, so its initialization should be delayed as much as possible.

Reply:
A reason I chose def would be sqliteFilePath can be used only once, so it saves the memory space a little bit by making it a def. 

The use of `lazy val` is usually for avoiding the same computation.

Reply:
I got it! Thanks for your answer!